cms/static/js/views/settings/grading.js

```diff
@@ -314,7 +315,7 @@ function(ValidatingView, _, $, ui, GraderView, StringUtils, HtmlUtils) {
         addNewGrade: function(e) {
             e.preventDefault();
             var gradeLength = this.descendingCutoffs.length; // cutoffs doesn't include fail/f so this is only the passing grades
-            if (gradeLength > 3) {
+            if (gradeLength > this.GRADES.length - 1) {
```

Comment:
@kaustavb12 could be a dumb question, but is there a possibility of `this.GRADES` being undefined if this line is invoked?

Eg. if options.gradeDesignations == 0 then these lines will trigger an error:

```javascript
this.setupGradeDesignations([])
this.addNewGrade()
```

Reply:
@keithgg

Great question.

We have already defined a [default value](https://github.com/open-craft/edx-platform/blob/a39905332d351d406ab0d961243676339392ffeb/cms/static/js/views/settings/grading.js#L168) of `this.GRADES`, therefore I feel the chances of `this.GRADES` being undefined is slim.

The place which is most likely to break with an erroneous value of `options.gradeDesignations` is the [length check](https://github.com/open-craft/edx-platform/blob/a39905332d351d406ab0d961243676339392ffeb/cms/static/js/views/settings/grading.js#L399) in the `setupGradeDesignations` method. Although, for most scenarios, such as passing an intereger, strings, empty arrays etc., this seems to be handled elegently by the browser and nothing breaks. Only explicitly passing `None/null` seems to break this.

To mitigate that, I [added a check](https://github.com/open-craft/edx-platform/blob/a39905332d351d406ab0d961243676339392ffeb/cms/static/js/views/settings/grading.js#L399) to verify that `gradeDesignations` is infact a valid array. This seems to handle those edge cases I mentioned above.